contracts/stake-lock/src/entry.rs

```diff
@@ -0,0 +1,75 @@
+//! Stake-lock
+
+// Import from `core` instead of from `std` since we are in no-std mode
+use core::result::Result;
+
+// Import heap related library from `alloc`
+// https://doc.rust-lang.org/alloc/index.html
+// use alloc::{vec, vec::Vec};
+
+// Import CKB syscalls and structures
+// https://nervosnetwork.github.io/ckb-std/riscv64imac-unknown-none-elf/doc/ckb_std/index.html
+use crate::ckb_std::{
+    ckb_constants::Source,
+    ckb_types::{bytes::Bytes, prelude::Unpack as CKBTypeUnpack},
+    high_level::{load_input_since, load_script},
+    since::Since,
+};
+
+use validator_utils::search_cells::{search_lock_hash, search_rollup_cell, search_rollup_state};
+
+use gw_types::{
+    packed::{StakeArgsLockReader, StakeLockArgs},
+    prelude::*,
+};
+
+use crate::error::Error;
+
+/// args: rollup_type_hash | stake lock args
+fn parse_lock_args() -> Result<([u8; 32], StakeLockArgs), Error> {
+    let mut rollup_type_hash = [0u8; 32];
+    let script = load_script()?;
+    let args: Bytes = script.args().unpack();
+    if args.len() < rollup_type_hash.len() {
+        return Err(Error::InvalidArgs);
+    }
+    rollup_type_hash.copy_from_slice(&args[..32]);
+    match StakeLockArgsReader::verify(&args, false) {
+        Ok(()) => Ok((
+            rollup_type_hash,
+            StakeLockArgs::new_unchecked(args.slice(32..)),
+        )),
+        Err(_) => Err(Error::InvalidArgs),
+    }
+}
+
+pub fn main() -> Result<(), Error> {
+    let (rollup_type_hash, lock_args) = parse_lock_args()?;
+
+    // read global state from rollup cell
+    let global_state = match search_rollup_state(&rollup_type_hash, Source::Dep)? {
+        Some(state) => state,
+        None => return Err(Error::RollupCellNotFound),
```

Comment:
If a staker tries to update the Rollup cell, then the Rollup cell actually exists in the tx.input, this pattern match will return an error. But according to the verification logic, we should try to unlock by Rollup cell after we failed to unlock by the user.

Reply:
Gonna fix it.